base/strings/search.jl

```diff
@@ -19,13 +22,13 @@ function findnext(pred::Fix2{<:Union{typeof(isequal),typeof(==)},<:AbstractChar}
     end
 end
 
-findfirst(pred::Fix2{<:Union{typeof(isequal),typeof(==)},<:Union{Int8,UInt8}}, a::ByteArray) =
+findfirst(pred::Fix2{<:Union{typeof(isequal),typeof(==)},T}, a::ByteArray{T}) where T<:Byte =
     nothing_sentinel(_search(a, pred.x))
 
-findnext(pred::Fix2{<:Union{typeof(isequal),typeof(==)},<:Union{Int8,UInt8}}, a::ByteArray, i::Integer) =
+findnext(pred::Fix2{<:Union{typeof(isequal),typeof(==)},T}, a::ByteArray{T}, i::Integer) where T<:Byte =
     nothing_sentinel(_search(a, pred.x, i))
 
-function _search(a::Union{String,ByteArray}, b::Union{Int8,UInt8}, i::Integer = 1)
+function _search(a::Union{String,ByteArray}, b::Byte, i::Integer = 1)
     if i < 1
         throw(BoundsError(a, i))
     end
```

Comment:
Alternatively?
```suggestion
    end
    typemin(eltype(a)) <= b <= typemax(eltype(a)) || return 0
```

Reply:
Thank you for your suggestion. That looks more elegant. However, I'd like to fix the problem in an upcoming pull request. I'm preparing an overhaul of `findnext` and friends to improve the consistency and performance, which can be found in https://github.com/bicycle1885/StringSearch.jl. This pull request is a stepping stone and the main purpose is to file an issue I found and add some test to make the next one less complicated.